targets/TARGET_NORDIC/TARGET_NRF5/cpu_uid_api.c

```diff
@@ -0,0 +1,59 @@
+/* 
+ * Copyright (c) 2017 Nordic Semiconductor ASA
+ * All rights reserved.
+ * 
+ * Redistribution and use in source and binary forms, with or without modification,
+ * are permitted provided that the following conditions are met:
+ * 
+ *   1. Redistributions of source code must retain the above copyright notice, this list 
+ *      of conditions and the following disclaimer.
+ *
+ *   2. Redistributions in binary form, except as embedded into a Nordic Semiconductor ASA 
+ *      integrated circuit in a product or a software update for such product, must reproduce 
+ *      the above copyright notice, this list of conditions and the following disclaimer in 
+ *      the documentation and/or other materials provided with the distribution.
+ *
+ *   3. Neither the name of Nordic Semiconductor ASA nor the names of its contributors may be 
+ *      used to endorse or promote products derived from this software without specific prior 
+ *      written permission.
+ *
+ *   4. This software, with or without modification, must only be used with a 
+ *      Nordic Semiconductor ASA integrated circuit.
+ *
+ *   5. Any software provided in binary or object form under this license must not be reverse 
+ *      engineered, decompiled, modified and/or disassembled. 
+ * 
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ * 
+ */
+
+#include "device.h"
+
+#if DEVICE_CPUUID
+
+#include "nrf.h"
+#include "cpu_uid_api.h"
+
+#define UID_WORDS       2
+
+void cpu_uid_get_uid(uint8_t *uid)
+{
+    int pos = 0;
+
+    for (int i = (UID_WORDS-1); i >= 0; --i) {
+        for (int j = 3; j >= 0; --j) {
+            uid[pos++] = (uint8_t)((NRF_FICR->DEVICEID[i] >> (j*8)) & 0xFF);
```

Comment:
For nRF51 and nRF52 families only bits [15:0] of DEVICEADDR[1] are used.

Reply:
ok that means for nRF51 and nRF52, the uid size is only 48 bits?
I will fix that tomorrow, when there is time ;)

Reply:
yes, correct

Reply:
I had a look into the nrf52832 product specification ... you are right about the DEVICEADDR registers .... but as you can see in the code, I read the DEVICEID registers, which seem to have no limitation...

Reply:
Yes - this is true. At last pleas add comment on that device ID is defined by these first 48-bites. I'm also wonder that it can be changed in future SoC - maybe it is better to set these unused bites to something.

Reply:
ok I still do not understand.

on the nrf52832, there are four registers:

- DEVICEID[0] addr.: 0x60
- DEVICEID[1] addr.: 0x64
- DEVICEADDR[0] addr.: 0xA4
- DEVICEADDR[1] addr.: 0xA8

Do you mean that DEVICEID is derived from DEVICEADDR ?
I can find nothing in the spec about that ( http://infocenter.nordicsemi.com/pdf/nRF52832_PS_v1.4.pdf )

Reply:
Excuse my, my fault - You have right from its was about DEVICEID, not DEVICEADDR. All my comment above are inwalida - DEVICEID represents 64-bit unique ID and your code is wrote properly.